mule-extensions-api/src/main/java/org/mule/runtime/extension/api/runtime/operation/OperationResult.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.runtime.operation;
+
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.api.metadata.MediaType;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+/**
+ * Represents the result of an operation execution. Extensions can use this class
+ * for cases in which the operation not only needs to return a value to be set
+ * on the message payload but also wants to specify message attributes and/or
+ * {@link MediaType}.
+ * <p>
+ * The {@link #getOutput()} value is always taken at face value, meaning that
+ * if it's {@code null}, then the message payload will also be nullified. However,
+ * if the {@link #getAttributes()} or {@link #getMediaType()} are {@link Optional#empty},
+ * then the runtime will interpret that as the operation not interested in changing
+ * those values, keeping the input message's attributes and/or media type untouched.
+ *
+ * @param <OUTPUT>     the generic type of the output value
+ * @param <ATTRIBUTES> the generic type of the message attributes
+ * @since 1.0
+ */
+public interface OperationResult<OUTPUT, ATTRIBUTES extends Serializable>
+{
+
+    /**
+     * Creates a new {@link Builder}
+     *
+     * @param <OUTPUT>     the generic type of the output value
+     * @param <ATTRIBUTES> the generic type of the message attributes
+     * @return a new {@link Builder}
+     */
+    static <OUTPUT, ATTRIBUTES extends Serializable> Builder<OUTPUT, ATTRIBUTES> builder()
```

Comment:
that's not how the convention is being used right now. I crafted this following the example of the MuleMessageBuilder